jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/selection/SelectionRow.java

```diff
@@ -48,10 +51,10 @@ public SelectionRow(EMovableType type, int count) {
 	@Override
 	public void drawAt(GLDrawContext gl) {
 		float width = getPosition().getWidth();
-		Image image =
-				SettlerImageMap.getInstance().getImageForSettler(type,
-						EMovableAction.NO_ACTION, EMaterialType.NO_MATERIAL,
-						EDirection.SOUTH_EAST, 0);
+		Image image = SettlerImageMap.getInstance().getImageForSettler(
+				new SettlerImageFlavor(type, EMovableAction.NO_ACTION, EMaterialType.NO_MATERIAL, EDirection.SOUTH_EAST),
```

Comment:
This is the main problem I see with your approach (and the reason why we did it the old way):

You are creating an object here that the JIT cannot (or at least in my tests with Android could not) inline. Mind that Android seems to be far worse at optimizing code than the normal Java JIT is.
This will make the young generation grow and thus create a lot of GC activity, since you might have thousands of settlers on one screen, especially when zooming out.
This is the reason why the drawing code attempts not to create any objects during painting phases.

We should test this and - if it is a problem - use a long instead of the SettlerImageFlavor (you can then create a static method that generates that unique long from the given parameters). Then use a library that supports long hash maps.

Reply:
You're right. I have overseen that fact. Maybe I can find a way to cache the SettlerImageFlavor instead of using a long since it would be the same approach. What do you think?

Reply:
The usual way to cache such objects is to put them in a hash table - You have a chicken and egg problem then ;-).

Generating the long is no big deal and should be pretty fast. A long -> Object hash map is probably the fastest you can get (in this case probably ~10 times faster than the way you are using now - at least that is my guess from my experience, might be more or less).

Reply:
I've looked at this again. The SettlerImageFlavor can be created once in the constructor and the overhead is simply gone.